src/shared/filesystemUtilities.ts

```diff
@@ -115,3 +117,33 @@ export function isInDirectory(d: string, p: string): boolean {
             : value === containedPathPieces[index]
     })
 }
+
+/**
+ * Returns `name.suffix` if it does not already exist in directory `dir`, else appends
+ * a number ("foo-1.txt", "foo-2.txt", etc.).
+ *
+ * To avoid excessive filesystem activity, if all filenames up to `max` exist,
+ * the function instead appends a random string.
```

Comment:
Is this a concern with `fs.readdir`/`readdirSync`? That pulls a list of folders and you can parse through that without additional fs acccess.

Reply:
Filesystem cache makes it a negligible concern in either case. The main point is the give the caller control, and avoid pathological cases where appending a number is not any more useful than a random string.

Reply:
If you're confident that potential 999 `existsSync` calls are equally expensive as a single `readdirSync` call, I'm fine with it. Technically `existsSync` is higher-fidelity if for some reason a user is creating new apps in the same directory at the same time, but it's moot because we're not immediately creating a directory when we find one that works (there's still a couple more steps in the workflow in which a clash could be made).

Reply:
`readdirSync` optimizes for the uncommon case. The overwhelmingly common case is that `existsSync` will be called once or twice, thus I opted to keep the code more obvious. Also, `readdirSync` would almost always do many comparisons, and potentially on large directories, only to search for a very specific name(s). I am not seeing the benefit of that tradeoff.